### ISSUE TITLE
4.0.0rc1 breaks Grammarly support

### ISSUE THREAD

**Issue by mattharrison** [jupyterlab/jupyterlab#14525]
## Description

Grammarly worked in the 4.0 betas but now it doesn't.

## Reproduce

If you have Grammarly and the extension, it no longer works with Jupyter lab.

## Expected behavior

The biggest motivator for me to use Lab vs Notebook was that Grammarly was working in the betas.

## Context

- Operating System and version: MacOS 12..1
- Browser and version: Chrome 112.0.5616.137
- JupyterLab version: 4.0.0.rc1

<details><summary>Troubleshoot Output</summary>
<pre>
.executable:
	/Users/matt/.envs/2023env/bin/python3.10

sys.version:
	3.10.11 (main, Apr  7 2023, 07:24:47) [Clang 14.0.0 (clang-1400.0.29.202)]

platform.platform():
	macOS-12.5.1-arm64-arm-64bit

which -a jupyter:
	/Users/matt/.envs/2023env/bin/jupyter

pip list:
	Package                  Version
	------------------------ ---------
	anyio                    3.6.2
	appnope                  0.1.3
	argon2-cffi              21.3.0
	argon2-cffi-bindings     21.2.0
	arrow                    1.2.3
	asttokens                2.2.1
	async-lru                2.0.2
	attrs                    22.2.0
	Babel                    2.12.1
	backcall                 0.2.0
	beautifulsoup4           4.12.1
	bleach                   6.0.0
	blis                     0.7.9
	catalogue                2.0.8
	certifi                  2022.12.7
	cffi                     1.15.1
	charset-normalizer       3.1.0
	click                    8.1.3
	cloudpickle              2.2.1
	comm                     0.1.3
	confection               0.0.4
	contourpy                1.0.7
	cycler                   0.11.0
	cymem                    2.0.7
	debugpy                  1.6.7
	decorator                5.1.1
	defusedxml               0.7.1
	en-core-web-sm           3.5.0
	et-xmlfile               1.1.0
	exceptiongroup           1.1.1
	executing                1.2.0
	fastjsonschema           2.16.3
	fonttools                4.39.3
	fqdn                     1.5.1
	future                   0.18.3
	hyperopt                 0.2.7
	idna                     3.4
	iniconfig                2.0.0
	ipykernel                6.22.0
	ipytest                  0.13.1
	ipython                  8.12.0
	ipython-genutils         0.2.0
	isoduration              20.11.0
	jedi                     0.18.2
	Jinja2                   3.1.2
	joblib                   1.2.0
	json5                    0.9.11
	jsonpointer              2.3
	jsonschema               4.17.3
	jupyter_client           8.1.0
	jupyter_core             5.3.0
	jupyter-events           0.6.3
	jupyter-lsp              2.1.0
	jupyter_server           2.5.0
	jupyter_server_terminals 0.4.4
	jupyterlab               4.0.0rc1
	jupyterlab-pygments      0.2.2
	jupyterlab_server        2.22.1
	jupytext                 1.14.5
	kiwisolver               1.4.4
	langcodes                3.3.0
	markdown-it-py           2.2.0
	MarkupSafe               2.1.2
	matplotlib               3.7.1
	matplotlib-inline        0.1.6
	mdit-py-plugins          0.3.5
	mdurl                    0.1.2
	mistune                  2.0.5
	murmurhash               1.0.9
	nbclassic                0.5.5
	nbclient                 0.7.3
	nbconvert                7.3.0
	nbformat                 5.8.0
	nest-asyncio             1.5.6
	networkx                 3.1
	notebook                 6.5.4
	notebook_shim            0.2.2
	numpy                    1.24.2
	openpyxl                 3.1.2
	packaging                23.0
	pandas                   2.0.1
	pandocfilters            1.5.0
	parso                    0.8.3
	pathy                    0.10.1
	pexpect                  4.8.0
	pickleshare              0.7.5
	Pillow                   9.5.0
	pip                      23.1.2
	platformdirs             3.2.0
	pluggy                   1.0.0
	polars                   0.17.5
	preshed                  3.0.8
	prometheus-client        0.16.0
	prompt-toolkit           3.0.38
	psutil                   5.9.4
	ptyprocess               0.7.0
	pure-eval                0.2.2
	py4j                     0.10.9.7
	pyarrow                  11.0.0
	pycparser                2.21
	pydantic                 1.10.7
	Pygments                 2.14.0
	pyparsing                3.0.9
	pyrsistent               0.19.3
	pytest                   7.3.1
	python-dateutil          2.8.2
	python-json-logger       2.0.7
	pytz                     2023.3
	PyYAML                   6.0
	pyzmq                    25.0.2
	requests                 2.28.2
	rfc3339-validator        0.1.4
	rfc3986-validator        0.1.1
	scikit-learn             1.2.2
	scipy                    1.10.1
	Send2Trash               1.8.0
	setuptools               67.2.0
	six                      1.16.0
	smart-open               6.3.0
	sniffio                  1.3.0
	soupsieve                2.4
	spacy                    3.5.1
	spacy-legacy             3.0.12
	spacy-loggers            1.0.4
	SQLAlchemy               2.0.9
	srsly                    2.4.6
	stack-data               0.6.2
	terminado                0.17.1
	thinc                    8.1.9
	threadpoolctl            3.1.0
	tinycss2                 1.2.1
	toml                     0.10.2
	tomli                    2.0.1
	tornado                  6.2
	tqdm                     4.65.0
	traitlets                5.9.0
	typer                    0.7.0
	typing_extensions        4.5.0
	tzdata                   2023.3
	uri-template             1.2.0
	urllib3                  1.26.15
	wasabi                   1.1.1
	wcwidth                  0.2.6
	webcolors                1.13
	webencodings             0.5.1
	websocket-client         1.5.1
	xgboost                  1.7.5

</pre>
</details>

**Comment by krassowski** [comment]
Can you easily try 4.0.0rc0 to narrow down when the change happened?

**Comment by mattharrison** [comment]
I reverted back to beta2 and it works. Will try rc0.

**Comment by mattharrison** [comment]
rc0 works with Grammarly

**Comment by JasonWeill** [comment]
4.0.0rc1 was released yesterday as JupyterLab 4.0.0, so I retitled this issue.
